From 4319d471cdc4d2775f6eb0f231702bc65933f6e7 Mon Sep 17 00:00:00 2001
From: Sunil Kumar Kori <skori@marvell.com>
Date: Tue, 16 Jul 2019 13:34:21 +0530
Subject: [PATCH 344/386] octeontx2-af: Set MAC address resource bit map to its
 default during FLR

cgx_lmac_addr_reset is responsible to reset LMAC to its default value.
During cgx_lmac_init, bit 0 is set of resource bit map
lmac->mac_to_index_bmap.bmap to reserve first entry of CGX table for
default MAC address.

But during FLR, same bit was being cleared instead setting to its default.
Because of this, default MAC address entry will be overwritten into CGX
table in consecutive MAC address add action.

Change-Id: I96c016aeb877ef16497cf00a0102128cd6a34cec
Signed-off-by: Sunil Kumar Kori <skori@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/12698
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 5d76826008ac..5533e69f4b65 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -237,8 +237,10 @@ int cgx_lmac_addr_reset(u8 cgx_id, u8 lmac_id)
 	u8 index = 0;
 	u64 cfg;
 
-	/*  Clear the reserved bit for default mac address */
-	clear_bit(0, lmac->mac_to_index_bmap.bmap);
+	/* Restore index 0 to its default init value as done during
+	 * cgx_lmac_init
+	 */
+	set_bit(0, lmac->mac_to_index_bmap.bmap);
 
 	index = lmac_id * lmac->mac_to_index_bmap.max + index;
 	cgx_write(cgx_dev, 0, (CGXX_CMRX_RX_DMAC_CAM0 + (index * 0x8)), 0);
-- 
2.17.1

